homeassistant/components/volkswagencarnet/__init__.py

```diff
@@ -0,0 +1,237 @@
+            return True
+
+        finally:
+            track_point_in_utc_time(hass, update, utcnow() + interval)
+
+    _LOGGER.info("Starting volkswagencarnet component")
+    return update(utcnow())
+
+
+class VolkswagenData:
+    """Hold component state."""
+
+    def __init__(self, config):
+        """Initialize the component state."""
+        self.vehicles = set()
+        self.instruments = set()
+        self.entities = {}
+        self.config = config[DOMAIN]
+        self.names = self.config.get(CONF_NAME)
+
+    def instrument(self, vin, component, attr):
+        """Return corresponding instrument."""
+        return next((instrument
+                     for instrument in self.instruments
+                     if instrument.vehicle.vin == vin and
+                     instrument.component == component and
+                     instrument.attr == attr), None)
+
+    def vehicle_name(self, vehicle):
+        """Provide a friendly name for a vehicle."""
+        if (vehicle.vin and vehicle.vin.lower() in self.names):
+            return self.names[vehicle.vin.lower()]
+        elif vehicle.vin:
+            return vehicle.vin
+        else:
+            return ''
+
+
+class VolkswagenEntity(Entity):
+    """Base class for all Volkswagen entities."""
+
+    def __init__(self, data, vin, component, attribute):#, attribute):
+        """Initialize the entity."""
+        self.data = data
+        self.vin = vin
+        self.component = component
+        self.attribute = attribute
+
+        self.data.entities[self.vin].append(self)
+
+    @property
+    def instrument(self):
+        """Return corresponding instrument."""
+        return self.data.instrument(self.vin, self.component, self.attribute)
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        if self.instrument.attr in ['battery_level', 'charging']:
+            return icon_for_battery_level(battery_level = self.instrument.state, charging = self.vehicle.charging)
+        else:
+            return self.instrument.icon
+
+    @property
+    def vehicle(self):
+        """Return vehicle."""
+        return self.instrument.vehicle
+
+    @property
+    def _entity_name(self):
+        return self.instrument.name
+
+    @property
+    def _vehicle_name(self):
+        return self.data.vehicle_name(self.vehicle)
+
+    @property
+    def name(self):
+        """Return full name of the entity."""
+        return '{} {}'.format(self._vehicle_name,self._entity_name)
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def assumed_state(self):
+        """Return true if unable to access real state of entity."""
+        return True
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return dict(self.instrument.attributes, model='{}/{}'.format(self.vehicle.model,self.vehicle.model_year))
```

Comment:
Please set the entity state attributes explicitly here. We want to know what they are and see how they are formatted. State attribute keys should be lowercase snakecase.